app/helpers/remote_execution_helper.rb

```diff
@@ -77,12 +77,11 @@ def template_invocation_status(task)
     end
   end
 
-  def template_invocation_actions(task, host)
-    if task.nil?
-      []
-    else
-      [display_link_if_authorized(_("Details"), hash_for_template_invocation_path(:id => task).merge(:auth_object => host, :permission => :view_hosts))]
-    end
+  def template_invocation_actions(task, host, job_invocation)
+    [
+      display_link_if_authorized(_('View host'), hash_for_host_path(host).merge(:auth_object => host, :permission => :view_hosts)),
+      display_link_if_authorized(_('Rerun on %s') % host.name, hash_for_rerun_job_invocation_path(:id => job_invocation, :host_ids => [ host.id ])),
```

Comment:
Is there a reason to show the hostname? Don't think we do that anywhere else in Foreman, do we?  I think it looks OK, but it'd be weird if there were multiple action items because you already know the host based on the row the button's in.